cpp/src/hash/hash_functions.cuh

```diff
@@ -189,4 +191,57 @@ struct IdentityHash<cudf::detail::wrapper<T,type_id>>
 template <typename Key>
 using default_hash = MurmurHash3_32<Key>;
 
+/**
+* @brief Specialization of MurmurHash3_32 operator for strings.
+*/
+template<>
+__forceinline__ 
+__host__ __device__ hash_value_type MurmurHash3_32<cudf::string_view>::operator()(const cudf::string_view& key) const
+{
+    const int len = (int)key.size_bytes();
+    const uint8_t* data = (const uint8_t*)key.data();
+    const int nblocks = len / 4;
```

Comment:
what if `len < 4`? Won't you end up hashing nothing?

Reply:
No. The `tail` section still executes
https://github.com/rapidsai/cudf/blob/637683d07cc6e867e487ec950c043855dc1c6db6/cpp/src/hash/hash_functions.cuh#L231

Reply:
Test case includes a single character as well.
https://github.com/rapidsai/cudf/blob/637683d07cc6e867e487ec950c043855dc1c6db6/cpp/tests/strings/hash_string.cu#L49

Reply:
Ah, okay. Makes sense.